app/models/pawn.rb

```diff
@@ -1,2 +1,22 @@
 class Pawn < Piece
-end
\ No newline at end of file
+  def valid_move?(new_rank, new_file)
+    rank_diff = new_rank - rank
+    file_diff = new_file - file
+
+    return false if is_move_malformed?(new_rank, new_file) || is_move_obstructed?(new_rank, new_file)
```

Comment:
I considered suggesting that in the context of `King#valid_move?`, but at the time, it seemed like the ideal flow from an efficiency standpoint would be the following, prohibiting the use of `super`:

- Call `is_move_malformed?`
- Execute type-specific logic
- Call `is_move_obstructed?`

Hilariously, I immediately violated that pattern here because the pawn logic is fairly complex and writing it out in terms of valid moves rather than failure conditions made the code much more clear. However, I was thinking we could return to it in the other versions of `valid_move?`.